pkg/pkgmgr/dpkg.go

```diff
@@ -159,26 +168,47 @@ func (dm *dpkgManager) probeDPKGStatus(ctx context.Context, toolImage string) er
 	busyBoxApplied := dm.config.ImageState.File(llb.Copy(busyBoxInstalled, "/bin/busybox", "/bin/busybox"))
 	mkFolders := busyBoxApplied.File(llb.Mkdir(resultsPath, 0o744, llb.WithParents(true)))
 
-	const probeTemplate = `/bin/busybox sh -c "if [ -f %[1]s ]; then cp %[1]s %[3]s ; fi && if [ -d %[2]s ]; then ls -1 %[2]s > %[4]s ; fi"`
-	probeCmd := fmt.Sprintf(probeTemplate, dpkgStatusPath, dpkgStatusFolder, resultsPath, filepath.Join(resultsPath, "status.d"))
-	probed := mkFolders.Run(llb.Shlex(probeCmd)).Root()
+	probed := mkFolders.
+		Run(llb.Args([]string{
+			`/bin/busybox`, `env`,
+			buildkit.Env("DPKG_STATUS_PATH", dpkgStatusPath),
+			buildkit.Env("RESULTS_PATH", resultsPath),
+			buildkit.Env("DPKG_STATUS_FOLDER", dpkgStatusFolder),
+			buildkit.Env("RESULT_STATUSD_PATH", filepath.Join(resultsPath, "status.d")),
+			buildkit.Env("DPKG_STATUS_IS_DIRECTORY", fmt.Sprintf("%d", DPKGStatusDirectory)),
+			buildkit.Env("DPKG_STATUS_IS_FILE", fmt.Sprintf("%d", DPKGStatusFile)),
+			buildkit.Env("DPKG_STATUS_IS_UNKNOWN", fmt.Sprintf("%d", DPKGStatusNone)),
+			buildkit.Env("STATUSD_OUTPUT_FILENAME", statusdOutputFilename),
+			`sh`, `-c`, `
+                status="$DPKG_STATUS_IS_UNKNOWN"
+                if [ -f "$DPKG_STATUS_PATH" ]; then
+                    status="$DPKG_STATUS_IS_FILE"
+                    cp "$DPKG_STATUS_PATH" "$RESULTS_PATH"
+                elif [ -d "$DPKG_STATUS_FOLDER" ]; then
+                    status="$DPKG_STATUS_IS_DIRECTORY"
+                    ls -1 "$DPKG_STATUS_FOLDER" > "$RESULT_STATUSD_PATH"
+                fi
+                echo -n "$status" > "${RESULTS_PATH}/${STATUSD_OUTPUT_FILENAME}"
+        `,
+		})).Root()
+
 	outState := llb.Diff(busyBoxApplied, probed)
-	if err := buildkit.SolveToLocal(ctx, dm.config.Client, &outState, dm.workingFolder); err != nil {
+	typeBytes, err := buildkit.ExtractFileFromState(ctx, dm.config.Client, &outState, filepath.Join(resultsPath, statusdOutputFilename))
```

Comment:
I am not sure why this is, but changing to using `.AddMount` results in the `buildkit.WithArrayFile` function not working. Those files just aren't in the container when the command is run. Any ideas why that might be?

Reply:
(this only applies to the corresponding spot in `rpm.go`)

Reply:
Note that its the contents of the mount only.
So if some file `foo` that gets written to `resultsPath/foo`, it will be in `/foo` for the result of `.AddMount`.

Reply:
It is the result of changing what is at the mount point.
If files are being written to that location first then yes they are being bind-mounted over with `llb.Scratch`.
Instead you can write the files to a different state (`resultsBase:= llb.Scratch().File(llb.Mkfile(...))`, then use `resultsBase in the `AddMount` instead of `llb.Scratch`).

Reply:
Done

Reply:
Did this not get pushed?